### ISSUE TITLE
Carousel: Items are not re-rendering(seeing old state) when parent state is updated

### ISSUE THREAD

**Issue by sraone** [eBay/ebayui-core#410]
# Bug Report
Items are not re-rendering(seeing old state) when parent state is updated

## Description
I am using a `carousel` items are bound from the state. When the state is replaced with new items, `carousel` still showing old items.

`template.marko`

```marko
<div w-bind>
    <button type="button" w-on-click="handleReloadClick">Reload</button>
    <ebay-carousel>
        <ebay-carousel-item for(item in data.items)>${item}</ebay-carousel-item>
    </ebay-carousel>
</div>
```

`index.js`

```javascript
module.exports = require('marko-widgets').defineComponent({
    template: require('./template.marko'),

    getInitialState() {
        return {
            items: [1, 2, 3, 4, 5, 6]
        }
    },

    handleReloadClick() {
        this.setState('items', [10, 20, 30]);
    }
});
```

## Screenshots

with initial state

![image](https://user-images.githubusercontent.com/1853563/45511675-7d2b6f80-b752-11e8-8a3c-1ea8b7cdbbb0.png)

with new state

![image](https://user-images.githubusercontent.com/1853563/45511750-af3cd180-b752-11e8-90b6-b337b7e70586.png)

But it should display `10, 20, 30`.

**Comment by yomed** [comment]
@DylanPiercey @mlrawlings It looks like the `renderBody` doesn't get updated correctly on the new input. Starting with `[1, 2]` and updating to `['a', 'b', 'c', 'd']` will result in `1 2 c d`.

**Comment by sraone** [comment]
@msendlakowski I don't think its marko issue because items are rendering as expected if I remove `carousel` component.

**Comment by msendlakowski** [comment]
Thanks.  I'll take a closer look.

**Comment by seangates** [comment]
@sraone Can you try the `1.1.0` branch and let us know if the change worked for you?

If you need help with how to test this branch let me know. I can assist.

**Comment by ianmcburnie** [comment]
Fixed in v1.1.0 branch.
